### PR TITLE
add a fast path to interp1d

### PR DESCRIPTION
**Pull request by ev-br**
As compared to `interp1d(.., kind='linear')`,  `numpy.interp` is 2x-10x faster, especially for sorted input. Hence delegate the former to the latter, in cases where numpy can handle it (specifically, `y` array must be real-valued and 1D).

The added complexity (and interp1d _is_ a bit of a mess) seems small and the perf boost is IMO worth it.

Also modify the interp1d benchmark to actually measure evaluation time.

I've checked the test coverage and there seems to be no need for any additional tests, the functionality added here is already covered.

### CODE REVIEW

**Comment by ev-br** [comment]
On branch: 

```
               =========== ========== ========== ========== ========== =========== ==========
               --                                        method                              
               ----------- ------------------------------------------------------------------
                n_samples    linear    nearest      zero     slinear    quadratic    cubic   
               =========== ========== ========== ========== ========== =========== ==========
                    10      33.15μs    41.94μs    82.83μs    82.62μs     83.06μs    85.64μs  
                    50      34.03μs    44.34μs    90.85μs    91.74μs     93.50μs    99.21μs  
                   100      35.53μs    49.44μs    105.95μs   108.62μs    116.15μs   126.09μs 
                   1000     65.46μs    145.52μs    1.82ms     1.76ms      1.74ms     1.89ms  
                  10000     363.29μs    1.42ms    152.25ms   151.72ms     failed     failed  
               =========== ========== ========== ========== ========== =========== ==========
```

On master:

```
               =========== ========== ========== ========== ========== =========== ==========
               --                                        method                              
               ----------- ------------------------------------------------------------------
                n_samples    linear    nearest      zero     slinear    quadratic    cubic   
               =========== ========== ========== ========== ========== =========== ==========
                    10      65.07μs    41.56μs    81.38μs    82.95μs     82.58μs    83.98μs  
                    50      68.08μs    45.06μs    91.21μs    91.84μs     94.00μs    101.47μs 
                   100      74.68μs    48.22μs    106.67μs   111.33μs    116.28μs   135.08μs 
                   1000     223.37μs   143.27μs    1.64ms     1.68ms      1.74ms     2.09ms  
                  10000      2.96ms     1.44ms    150.83ms   151.59ms     failed     failed  
               =========== ========== ========== ========== ========== =========== ==========
```

File: benchmarks/benchmarks/interpolate.py [hunk 1]
```diff
@@ -93,16 +93,18 @@ def time_evaluation(self, n_grids, method):
...
 
     def setup(self, n_samples, method):
         self.x = np.arange(n_samples)
         self.y = np.exp(-self.x/3.0)
+        self.interpolator = interpolate.interp1d(self.x, self.y, kind=method)
+        self.xp = np.linspace(self.x[0], self.x[-1], 4*n_samples)
 
     def time_interpolate(self, n_samples, method):
-        interpolate.interp1d(self.x, self.y, kind=method)
+        self.interpolator(self.xp)
```
Comment (pv) [inline benchmarks/benchmarks/interpolate.py]: Maybe one benchmark for construction and a second one for evaluation?
The construction overhead may in principle be also interesting to track.

Comment (ev-br) [inline benchmarks/benchmarks/interpolate.py]: Done.

**Comment by pv** [comment]
Thanks, merged.

**Pull request merged**